src/Database/Type/DateTimeType.php

```diff
@@ -179,18 +194,21 @@ public function toPHP($value, DriverInterface $driver)
             return null;
         }
 
-        $instance = clone $this->_datetimeInstance;
+        $class = $this->_className;
         if (is_int($value)) {
-            $instance = $instance->setTimestamp($value);
+            $instance = new $class('@' . $value);
         } else {
             if (strpos($value, '0000-00-00') === 0) {
                 return null;
             }
-            $instance = $instance->modify($value);
+            $instance = new $class($value, $this->dbTimezone);
```

Comment:
Caching off the DateTimeZone instance removes the overhead of creating new DateTime instances here.